### ISSUE TITLE
Ancillary Variables - Fix concatenate, merge

### ISSUE THREAD

**Issue by trexfeathers** [SciTools/iris#3552]
See #3422  [this comment](https://github.com/SciTools/iris/pull/3422#issuecomment-545480015) and #3483  [this comment](https://github.com/SciTools/iris/issues/3483#issuecomment-545503186)

Ancillary variable support should not be released without ensuring `concatenate` and `merge` handle AV's correctly

---
### Acceptance Criteria
All to be shown working with tests
  - [x] cubes with matching ancillary variables can merge to a result containing the ancillary
  - [x] likewise.., cubes with matching ancillary variables (not along the axis of concatenation) can concatenate to a result containing the ancillary
  - [x] a cube with an ancillary variables and one without (a matching one) will not merge;
     * ... and a descriptive error message is provided, similar to other merge failures
     * ... and the special extra message provided by "merge_cube" describes the precise difficulty
  - [x] cubes with differing ancillary variables (i.e. same names but some difference) will not merge;
     * ... likewise, message provisions as above
  - [x] likewise.., cubes with/without ancillary variables will not concatenate, and produce appropriate messages
  - [x] likewise.., cube with differing ancillary variables will not concatenate, and produce appropriate messages
  - [x] cubes with similar ancillary variables which span the axis of concatenation, will concatenate to form a new ancillary variable which spans the concatenated cube
 - [x] as all above with cell measures

In short, ancillary variables and cell measures will behave the same as (non-scalar) aux coords.

----
(@pp-mo 2019-11-22) Notes on changes I just made to this ...
  * added the "experience-needed: High" label because the existing merge+concatenate code is quite complex, so getting it + proposing changes with confidence is hard !
  * added some initial thoughts on acceptance criteria : not necessarily complete; I just updated this while working on something else
